src/test/java/wedlog/address/logic/parser/VendorFilterCommandParserTest.java

```diff
@@ -0,0 +1,97 @@
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import wedlog.address.logic.commands.VendorFilterCommand;
+import wedlog.address.logic.parser.exceptions.ParseException;
+import wedlog.address.model.person.AddressPredicate;
+import wedlog.address.model.person.EmailPredicate;
+import wedlog.address.model.person.NamePredicate;
+import wedlog.address.model.person.PhonePredicate;
+import wedlog.address.model.person.Vendor;
+
+class VendorFilterCommandParserTest {
+    private VendorFilterCommandParser parser = new VendorFilterCommandParser();
+
+    @Test
+    public void parse_invalidValue_failure() {
+        // no predicates
+        assertParseFailure(parser, " ", String.format(MESSAGE_NO_PREFIX_FOUND,
+                VendorFilterCommand.MESSAGE_USAGE));
+
+        // non empty preamble
+        assertParseFailure(parser, PREAMBLE_NON_EMPTY,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, VendorFilterCommand.MESSAGE_USAGE));
+
+        // empty name
+        assertParseFailure(parser, " " + PREFIX_NAME, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        "Cannot filter for empty compulsory field."));
+
+        // repeated name
+        assertThrows(ParseException.class, () -> parser.parse(" n/val n/val"));
+
+        // repeated phone
+        assertThrows(ParseException.class, () -> parser.parse(PHONE_DESC_VAL + PHONE_DESC_VAL));
+
+        // repeated email
+        assertThrows(ParseException.class, () -> parser.parse(EMAIL_DESC_VAL + EMAIL_DESC_VAL));
+
+        // repeated address
+        assertThrows(ParseException.class, () -> parser.parse(" a/Jurong a/West"));
+    }
+
+    @Test
+    public void parse_onlyOneFieldPresent_success() throws ParseException {
+        // name present
+        VendorFilterCommand vendorFilterCommand = parser.parse(" n/val");
+        List<Predicate<? super Vendor>> predicateList = Collections.singletonList(
+                new NamePredicate(Collections.singletonList("val")));
+        assertEquals(vendorFilterCommand, new VendorFilterCommand(predicateList));
+
+        // phone present
+        vendorFilterCommand = parser.parse(PHONE_DESC_VAL);
+        predicateList = Collections.singletonList(new PhonePredicate(Collections.singletonList(VALID_PHONE_VAL)));
+        assertEquals(vendorFilterCommand, new VendorFilterCommand(predicateList));
+
+        // email present
+        vendorFilterCommand = parser.parse(EMAIL_DESC_VAL);
+        predicateList = Collections.singletonList(new EmailPredicate(Collections.singletonList(VALID_EMAIL_VAL)));
+        assertEquals(vendorFilterCommand, new VendorFilterCommand(predicateList));
+
+        // address present
+        vendorFilterCommand = parser.parse(" a/jurong");
+        predicateList = Collections.singletonList(new AddressPredicate(Collections.singletonList("jurong")));
+        assertEquals(vendorFilterCommand, new VendorFilterCommand(predicateList));
+    }
```

Comment:
The the tests here for single present fields in this file looks good! Though, I think a test for RSVP present would help cover checks for if only RSVP is present too. (This is observed in other places as well)

Reply:
Vendors do not have RSVP fields, so I don't think that should be tested here. Where else was this issue observed?